python/ray/autoscaler/autoscaler.py

```diff
@@ -199,6 +202,62 @@ def _info(self):
         }
 
 
+class NodeLauncher(threading.Thread):
+    def __init__(self, queue, pending, config, *args, **kwargs):
+        self.queue = queue
+        self.pending = pending
+        self.provider = get_node_provider(config["provider"],
```

Comment:
It might be worth delaying the provider init until needed.

Reply:
Agreed, I'll make that change.